lib/tasks/cleanup.rake

```diff
@@ -124,4 +124,56 @@ namespace :cleanup do
 
     puts "Removed #{total} contributions; deleted #{deleted} publications, updated #{updated} publications."
   end
+
+ desc 'Remove all new contributions and related publications with no other contributions after a specific date -- rarely used for bad harvests'
+  # Use case: a harvest goes awry and you need to cleanup the contributions/publications that resulted.  This should be rare.
+  #  Input a date and a provenance.  Any publications created with that provenance after or equal to that date that also had a single
+  #  contribution created and whose status is 'new' (to avoid manually added publications) after that date will be destroyed, along with the related source record.
+  #  NOTE: you should only do this if CAP has not yet updated on their end or else you will be out of sync.
+  # RAILS_ENV=production bundle exec rake cleanup:harvest['May 6 2019','pubmed'] # will remove all contributions in the 'new' state after the given date and with the given provenance
+  task :harvest, [:start_date, :provenance] => :environment do |_t, args|
+
+    start_date = args[:start_date]
```

Comment:
Can this code be moved into a class that the rake task calls?  That would make it easier to test

Reply:
makes sense, sort of hoping that is something that only needs to be run this one time though, and also just looking for general feedback on this approach to cleanup of potential bad publications having been created